format/tzif/tzif.go

```diff
@@ -0,0 +1,205 @@
+	}
+
+	s.Sym = time.Unix(i, 0).UTC().Format(time.RFC3339)
+	return s, nil
+}
+
+var unixTimeToStr unixTimeToStrMapper
+
+func decodeTZifDataBlock(d *decode.D, h tzifHeader, decodeAsVer int, name string) {
+	timeSize := 8 * 8
+	if decodeAsVer == 1 {
+		timeSize = 4 * 8
+	}
+
+	d.FieldStruct(name, func(d *decode.D) {
+
+		d.FieldArray("transition_times", func(d *decode.D) {
+			for i := uint32(0); i < h.timecnt; i++ {
+				t := d.FieldS("transition_time", timeSize, unixTimeToStr)
+				if t < -576460752303423488 {
+					d.Fatalf("transition time value should be at least -2^59 (-576460752303423488), but: %d (%0#16x)", t, t)
+				}
+			}
+		})
+
+		d.FieldArray("transition_types", func(d *decode.D) {
+			for i := uint32(0); i < h.timecnt; i++ {
+				t := uint8(d.FieldU8("transition_type"))
+				if uint32(t) >= h.typecnt {
+					d.Fatalf("transition type must be in the range [0, %d]", h.typecnt-1)
+				}
+			}
+		})
+
+		d.FieldArray("local_time_type_records", func(d *decode.D) {
+			for i := uint32(0); i < h.typecnt; i++ {
+				d.FieldStruct("local_time_type", func(d *decode.D) {
+					d.FieldS32("utoff", d.AssertSRange(-89999, 93599))
+					d.FieldU8("dst", d.AssertU(0, 1))
+					d.FieldU8("idx", d.AssertURange(0, uint64(h.charcnt)-1))
+				})
+			}
+		})
+
+		d.FieldArray("time_zone_designations", func(d *decode.D) {
+			i := int(h.charcnt)
+			for {
+				s := d.FieldUTF8Null("time_zone_designation")
+				i -= len(s) + 1
+				if i <= 0 {
+					break
+				}
+			}
+		})
+
+		d.FieldArray("leap_second_records", func(d *decode.D) {
+			prevOccur := int64(0)
+			prevCorr := int64(0)
+
+			for i := uint32(0); i < h.leapcnt; i++ {
+				d.FieldStruct("leap_second_record", func(d *decode.D) {
+					occur := d.FieldS("occur", timeSize, unixTimeToStr)
+					corr := d.FieldS32("corr")
+
+					if i == 0 && occur < 0 {
+						d.Fatalf("the first value of occur must be nonnegative")
+					}
+					if i > 0 && occur-prevOccur < 2419199 {
+						d.Fatalf("occur must be at least 2419199 greater than the previous value")
+					}
+					if i == 0 && corr != 1 && corr != -1 {
+						d.Fatalf("the first value of corr must be either 1 or -1")
+					}
+					diff := corr - prevCorr
+					if i > 0 && diff != 1 && diff != -1 {
+						d.Fatalf("corr must differ by exactly 1 from the previous value: diff = %d, current corr = %d, previous corr = %d", diff, corr, prevCorr)
+					}
+
+					prevOccur = occur
+					prevCorr = corr
+				})
+			}
+		})
+
+		d.FieldArray("standard_wall_indicators", func(d *decode.D) {
+			for i := uint32(0); i < h.isstdcnt; i++ {
+				d.FieldU8("standard_wall_indicator", d.AssertU(0, 1))
+			}
+		})
+
+		d.FieldArray("ut_local_indicators", func(d *decode.D) {
+			for i := uint32(0); i < h.isutcnt; i++ {
+				d.FieldU8("ut_local_indicator", d.AssertU(0, 1))
+			}
+		})
+	})
+}
+
+func decodeTZifFooter(d *decode.D) {
+	d.FieldStruct("footer", func(d *decode.D) {
+		d.FieldU8("nl1", d.AssertU(0x0a))
+		i := 1
+		for {
+			b := d.PeekBytes(i)
```

Comment:
Maybe can use `d.PeekFindByte` somehow? the is also `d.PeekFind` if want some complicated condition or `d.FieldRE` that is very powerful but i haven't used it in any public decoder yet, not sure about the api.

Reply:
Oh `d.PeekFindByte()` is what I was looking for :joy: 
Updated @ [81b0b09](https://github.com/wader/fq/pull/498/commits/81b0b09754e851c5326bfa0c54db1f43d890ff58)